### PR TITLE
placement: support RuleGroup properties

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?

Tracking issue: https://github.com/pingcap/pd/issues/2721

### What is changed and how it works?

Add RuleGroup and consider it when calculating rule to fit a region. 
(Before complete the feature fully, it won't take any affect)

### Check List

Tests
- Unit test

### Release note
- No release note

### CODE REVIEW

File: server/schedule/placement/rule.go [hunk 1]
```diff
@@ -54,26 +54,46 @@ type Rule struct {
...
+func (r *Rule) groupIndex() int {
+	if r.group != nil {
+		return r.group.Index
+	}
+	return 0
+}
+
+// RuleGroup defines properties of a rule group.
+type RuleGroup struct {
+	ID       string `json:"id,omitempty"`
```
Comment (JmPotato) [inline server/schedule/placement/rule.go:85]: Does this field conflict with `Rule. GroupID`?

Comment (disksing) [reply server/schedule/placement/rule.go:85]: Rules and RuleGroups will be maintained and stored separately. ID is used to associate them together. (consider it as a foreign key)

Comment (JmPotato) [reply server/schedule/placement/rule.go:85]: I got it.

File: server/schedule/placement/rule_manager.go [hunk 2]
```diff
@@ -365,3 +367,14 @@ func (m *RuleManager) Batch(todo []RuleOp) error {
 	log.Info("placement rules updated", zap.String("batch", fmt.Sprint(todo)))
 	return nil
 }
+
+func (m *RuleManager) rebuildRuleList() (ruleList, error) {
+	for _, r := range m.rules {
+		r.group = m.groups[r.GroupID]
```
Comment (JmPotato) [inline server/schedule/placement/rule_manager.go:373]: When will the `RuleManager.Group` (i.e `m.groups`) be initialized?

Comment (disksing) [reply server/schedule/placement/rule_manager.go:373]: RuleManager.Initialize(): load from etcd // not included in this PR

Comment (JmPotato) [reply server/schedule/placement/rule_manager.go:373]: Ok.

**Pull request merged**